Update resolve dependency to 1.12.0

The following tests fail with resolve 1.12.0

```
$ mocha -R spec


  ✓ shim found
  ✓ core shim not found
  ✓ shim found
  ✓ core shim not found
  ✓ module to implicit extension
  ✓ implicit extension to implicit extension
  ✓ implicit extension to implicit extension
  ✓ explicit extension to explicit extension
  ✓ implicit extension to explicit extension
  ✓ module implicit extension to explicit extension
  ✓ module implicit extension to explicit extension
  ✓ false file
  ✓ false module
  ✓ false file
  ✓ false module
  ✓ false expand path
  ✓ local
  ✓ local
  ✓ local
  ✓ index.js of module dir
  ✓ alternate main
  ✓ string browser field as main
  ✓ string browser field as main - require subfile
  ✓ object browser field as main
  ✓ object browser field replace file
  ✓ object browser field replace file - no paths
  ✓ replace module in browser field object
  ✓ index.js of module dir
  ✓ alternate main
  ✓ string browser field as main
  ✓ string browser field as main - require subfile
  ✓ object browser field as main
  ✓ object browser field as main
  ✓ object browser field replace file
  ✓ test foobar -> module-b replacement
  ✓ test ./x -> ./y replacement
  ✓ test core -> module-c replacement
  ✓ test foobar -> module-b replacement with transform
  ✓ test foobar -> module-i replacement with transform in replacement
  ✓ object browser field replace file - no paths
  ✓ replace module in browser field object
  ✓ override engine shim
  ✓ alt-browser field
  1) index.js of module dir
  ✓ alternate main
  ✓ string browser field as main
  ✓ string browser field as main - require subfile
  ✓ string alt browser field as main - require subfile
  ✓ object browser field as main
  ✓ object browser field as main
  2) deep module reference mapping
  3) deep module reference mapping without file extension - .js
  4) deep module reference mapping without file extension - .json
  ✓ object browser field replace file
  ✓ test foobar -> module-b replacement
  ✓ test core -> module-c replacement
  ✓ test core -> module-c replacement with alt browser
  ✓ test foobar -> module-b replacement with transform
  ✓ test ./x -> ./y replacement
  ✓ test foobar -> module-i replacement with transform in replacement
  ✓ object browser field replace file - no paths
  ✓ replace module in browser field object
  ✓ override engine shim
  ✓ alt-browser field
  5) alt-browser deep module reference mapping
  ✓ alt-browser fallback to "browser" on deps of deps
  6) not fail on accessing path name defined in Object.prototype

  61 passing (198ms)
  6 failing

  1) index.js of module dir:
     Uncaught AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- {
-   main: 'fixtures'
- }
+ undefined
      at test/modules.js:11:16
      at index.js:269:13
      at node_modules/resolve/lib/async.js:93:25
      at maybeUnwrapSymlink (node_modules/resolve/lib/async.js:35:9)
      at node_modules/resolve/lib/async.js:89:24
      at ondir (node_modules/resolve/lib/async.js:264:27)
      at onex (node_modules/resolve/lib/async.js:161:32)
      at node_modules/resolve/lib/async.js:11:20
      at FSReqWrap.oncomplete (fs.js:155:5)

  2) deep module reference mapping:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-l/direct' from '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures'
      at test/modules.js:95:16
      at index.js:265:24
      at node_modules/resolve/lib/async.js:99:17
      at node_modules/resolve/lib/async.js:97:35
      at processDirs (node_modules/resolve/lib/async.js:244:39)
      at isdir (node_modules/resolve/lib/async.js:251:32)
      at node_modules/resolve/lib/async.js:23:69
      at FSReqWrap.oncomplete (fs.js:154:21)

  3) deep module reference mapping without file extension - .js:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-n/foo' from '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures'
      at test/modules.js:106:16
      at index.js:265:24
      at node_modules/resolve/lib/async.js:99:17
      at node_modules/resolve/lib/async.js:97:35
      at processDirs (node_modules/resolve/lib/async.js:244:39)
      at isdir (node_modules/resolve/lib/async.js:251:32)
      at node_modules/resolve/lib/async.js:23:69
      at FSReqWrap.oncomplete (fs.js:154:21)

  4) deep module reference mapping without file extension - .json:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-n/bar' from '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures'
      at test/modules.js:113:16
      at index.js:265:24
      at node_modules/resolve/lib/async.js:99:17
      at node_modules/resolve/lib/async.js:97:35
      at processDirs (node_modules/resolve/lib/async.js:244:39)
      at isdir (node_modules/resolve/lib/async.js:251:32)
      at node_modules/resolve/lib/async.js:23:69
      at FSReqWrap.oncomplete (fs.js:154:21)

  5) alt-browser deep module reference mapping:

      Uncaught AssertionError [ERR_ASSERTION]: '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/alt-browser-field/direct.js' == '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/alt-browser-field/chromeapp-direct.js'
      + expected - actual

      -/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/alt-browser-field/direct.js
      +/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/alt-browser-field/chromeapp-direct.js
      
      at test/modules.js:292:16
      at index.js:269:13
      at node_modules/resolve/lib/async.js:93:25
      at maybeUnwrapSymlink (node_modules/resolve/lib/async.js:35:9)
      at node_modules/resolve/lib/async.js:89:24
      at onfile (node_modules/resolve/lib/async.js:258:27)
      at onex (node_modules/resolve/lib/async.js:161:32)
      at node_modules/resolve/lib/async.js:11:20
      at FSReqWrap.oncomplete (fs.js:155:5)

  6) not fail on accessing path name defined in Object.prototype:

      Uncaught AssertionError [ERR_ASSERTION]: 'toString' == '/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/toString/index.js'
      + expected - actual

      -toString
      +/home/vagrant/forge/debian/git/js-team/node-browser-resolve/test/fixtures/node_modules/toString/index.js
      
      at test/modules.js:317:16
      at index.js:269:13
      at node_modules/resolve/lib/async.js:87:38
      at ondir (node_modules/resolve/lib/async.js:264:27)
      at onex (node_modules/resolve/lib/async.js:161:32)
      at node_modules/resolve/lib/async.js:11:20
      at FSReqWrap.oncomplete (fs.js:155:5)
```

New failures are:
```
  1) index.js of module dir:
     Uncaught AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual                                                                                                                                                                 

- {                                                                                                                                                                                 
-   main: 'fixtures'                                                                                                                                                                
- }                                                                                                                                                                                 
+ undefined                                                                                                                                                                         
      at test/modules.js:11:16                                                                                                                                                      
      at index.js:269:13                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:93:25                                                                                                                               
      at maybeUnwrapSymlink (/usr/share/nodejs/resolve/lib/async.js:35:9)                                                                                                           
      at /usr/share/nodejs/resolve/lib/async.js:89:24                                                                                                                               
      at ondir (/usr/share/nodejs/resolve/lib/async.js:270:27)                                                                                                                      
      at onex (/usr/share/nodejs/resolve/lib/async.js:161:32)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:11:20                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:154:5)                                                                                                                                         

  2) deep module reference mapping:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-l/direct' from '/<<PKGBUILDDIR>>/test/fixtures'
      at test/modules.js:95:16                                                                                                                                                      
      at index.js:265:24                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:99:17                                                                                                                               
      at /usr/share/nodejs/resolve/lib/async.js:97:35                                                                                                                               
      at processDirs (/usr/share/nodejs/resolve/lib/async.js:250:39)                                                                                                                
      at ondir (/usr/share/nodejs/resolve/lib/async.js:271:13)                                                                                                                      
      at load (/usr/share/nodejs/resolve/lib/async.js:137:43)                                                                                                                       
      at onex (/usr/share/nodejs/resolve/lib/async.js:162:17)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:13:69                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:153:21)                                                                                                                                        

  3) deep module reference mapping without file extension - .js:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-n/foo' from '/<<PKGBUILDDIR>>/test/fixtures'
      at test/modules.js:106:16                                                                                                                                                     
      at index.js:265:24                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:99:17                                                                                                                               
      at /usr/share/nodejs/resolve/lib/async.js:97:35                                                                                                                               
      at processDirs (/usr/share/nodejs/resolve/lib/async.js:250:39)                                                                                                                
      at ondir (/usr/share/nodejs/resolve/lib/async.js:271:13)                                                                                                                      
      at load (/usr/share/nodejs/resolve/lib/async.js:137:43)                                                                                                                       
      at onex (/usr/share/nodejs/resolve/lib/async.js:162:17)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:13:69                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:153:21)                                                                                                                                        

  4) deep module reference mapping without file extension - .json:
     Uncaught AssertionError [ERR_ASSERTION]: ifError got unwanted exception: Cannot find module 'module-n/bar' from '/<<PKGBUILDDIR>>/test/fixtures'
      at test/modules.js:113:16                                                                                                                                                     
      at index.js:265:24                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:99:17                                                                                                                               
      at /usr/share/nodejs/resolve/lib/async.js:97:35                                                                                                                               
      at processDirs (/usr/share/nodejs/resolve/lib/async.js:250:39)                                                                                                                
      at ondir (/usr/share/nodejs/resolve/lib/async.js:271:13)                                                                                                                      
      at load (/usr/share/nodejs/resolve/lib/async.js:137:43)                                                                                                                       
      at onex (/usr/share/nodejs/resolve/lib/async.js:162:17)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:13:69                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:153:21)                                                                                                                                        

  5) alt-browser deep module reference mapping:

      Uncaught AssertionError [ERR_ASSERTION]: '/<<PKGBUILDDIR>>/test/fixtures/node_modules/alt-browser-field/direct.js' == '/<<PKGBUILDDIR>>/test/fixtures/node_modules/alt-browser-field/chromeapp-direct...                                                                                                                                                          
      + expected - actual

      -/<<PKGBUILDDIR>>/test/fixtures/node_modules/alt-browser-field/direct.js
      +/<<PKGBUILDDIR>>/test/fixtures/node_modules/alt-browser-field/chromeapp-direct.js
      
      at test/modules.js:292:16                                                                                                                                                     
      at index.js:269:13                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:93:25                                                                                                                               
      at maybeUnwrapSymlink (/usr/share/nodejs/resolve/lib/async.js:35:9)                                                                                                           
      at /usr/share/nodejs/resolve/lib/async.js:89:24                                                                                                                               
      at onfile (/usr/share/nodejs/resolve/lib/async.js:264:27)                                                                                                                     
      at onex (/usr/share/nodejs/resolve/lib/async.js:161:32)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:11:20                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:154:5)                                                                                                                                         

  6) not fail on accessing path name defined in Object.prototype:
     Uncaught AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual                                                                                                                                                                 

- {                                                                                                                                                                                 
-   main: 'fixtures'                                                                                                                                                                
- }                                                                                                                                                                                 
+ undefined                                                                                                                                                                         
      at test/modules.js:318:16                                                                                                                                                     
      at index.js:269:13                                                                                                                                                            
      at /usr/share/nodejs/resolve/lib/async.js:93:25                                                                                                                               
      at maybeUnwrapSymlink (/usr/share/nodejs/resolve/lib/async.js:35:9)                                                                                                           
      at /usr/share/nodejs/resolve/lib/async.js:89:24                                                                                                                               
      at ondir (/usr/share/nodejs/resolve/lib/async.js:270:27)                                                                                                                      
      at onex (/usr/share/nodejs/resolve/lib/async.js:161:32)                                                                                                                       
      at /usr/share/nodejs/resolve/lib/async.js:11:20                                                                                                                               
      at FSReqWrap.oncomplete (fs.js:154:5)     
```

--- a/index.js
+++ b/index.js
@@ -206,7 +206,7 @@ function build_resolve_opts(opts, base) {
     return opts;
 }
 
-function resolve(id, opts, callback) {
+function resolve(id, opts, cb) {
 
     // opts.filename
     // opts.paths
@@ -215,24 +215,6 @@ function resolve(id, opts, callback) {
 
     opts = opts || {};
     opts.filename = opts.filename || '';
-
-    var cb = function(err, path, pkg) {
-        fs.stat(path, function(notPath) {
-            if (notPath) {
-                callback(err, path, pkg);
-            }
-            else {
-                fs.realpath(path, function(notReal, real) {
-                    if (notReal) {
-                        callback(err, path, pkg);
-                    }
-                    else {
-                        callback(err, real, pkg);
-                    }
-                });
-            }
-        });
-    }
 
     var base = path.dirname(opts.filename);
 

--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
   "author": "Roman Shtylman <shtylman@gmail.com>",
   "license": "MIT",
   "dependencies": {
-    "resolve": "1.1.7"
+    "resolve": "^1.17.0"
   },
   "devDependencies": {
     "mocha": "^2.5.3"

--- a/test/modules.js
+++ b/test/modules.js
@@ -8,7 +8,7 @@ test('index.js of module dir', function(done) {
     resolve('module-a', { paths: [ fixtures_dir ], package: { main: 'fixtures' } }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-a/index'));
-        assert.strictEqual(pkg, undefined);
+        assert.strictEqual(pkg.main, 'fixtures');
         done();
     });
 });
@@ -70,7 +70,7 @@ test('string alt browser field as main - require subfile', function(done) {
 // one of the keys replaces the main file
 // this would be done if the user needed to replace main and some other module
 test('object browser field as main', function(done) {
-    resolve('module-d', { paths: [ fixtures_dir ], package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('module-d', { paths: [ fixtures_dir ] }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-d/browser'));
         assert.equal(pkg.main, './browser.js');
@@ -82,7 +82,7 @@ test('object browser field as main', function(done) {
 // one of the keys replaces the main file
 // however the main has no prefix and browser uses ./ prefix for the same file
 test('object browser field as main', function(done) {
-    resolve('module-k', { paths: [ fixtures_dir ], package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('module-k', { paths: [ fixtures_dir ] }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-k/browser'));
         assert.equal(pkg.main, './browser.js');
@@ -91,7 +91,7 @@ test('object browser field as main', function(done) {
 });
 
 test('deep module reference mapping', function(done) {
-    resolve('module-l/direct', { basedir: __dirname + '/fixtures', package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('module-l/direct', { basedir: __dirname + '/fixtures' }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-l/browser-direct'));
         assert.equal(pkg.main, './browser.js');
@@ -102,14 +102,14 @@ test('deep module reference mapping', function(done) {
 // package.json has browser field as object
 // test that file resolves even though the file extension is omitted
 test('deep module reference mapping without file extension - .js', function(done) {
-    resolve('module-n/foo', { basedir: __dirname + '/fixtures', package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('module-n/foo', { basedir: __dirname + '/fixtures' }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-n/browser-foo'));
         done();
     });
 });
 test('deep module reference mapping without file extension - .json', function(done) {
-    resolve('module-n/bar', { basedir: __dirname + '/fixtures', package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('module-n/bar', { basedir: __dirname + '/fixtures' }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/module-n/browser-bar'));
         done();
@@ -285,7 +285,6 @@ test('alt-browser field', function(done) {
 test('alt-browser deep module reference mapping', function(done) {
     resolve('alt-browser-field/direct', {
         basedir: __dirname + '/fixtures',
-        package: { main: 'fixtures' },
         browser: 'chromeapp'
     }, function(err, path, pkg) {
         assert.ifError(err);
@@ -312,10 +311,9 @@ test('alt-browser fallback to "browser" on deps of deps', function(done) {
 });
 
 test('not fail on accessing path name defined in Object.prototype', function (done) {
-    resolve('toString', { paths: [ fixtures_dir ], package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('toString', { paths: [ fixtures_dir ] }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/node_modules/toString/index'));
-        assert.strictEqual(pkg, undefined);
         done();
     });
 });
@@ -334,10 +332,9 @@ test('respect symlinks', function (done) {
     //     - node_modules
     //       - symlink to x
     //
-    resolve('linked', { paths: [ fixtures_dir + '/linker/node_modules' ], package: { main: 'fixtures' } }, function(err, path, pkg) {
+    resolve('linked', { paths: [ fixtures_dir + '/linker/node_modules' ] }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/linked/index'));
-        assert.strictEqual(pkg, undefined);
         done();
     });
 });

--- a/test/modules.js
+++ b/test/modules.js
@@ -332,7 +332,7 @@ test('respect symlinks', function (done) {
     //     - node_modules
     //       - symlink to x
     //
-    resolve('linked', { paths: [ fixtures_dir + '/linker/node_modules' ] }, function(err, path, pkg) {
+    resolve('linked', { paths: [ fixtures_dir + '/linker/node_modules' ], preserveSymlinks: false }, function(err, path, pkg) {
         assert.ifError(err);
         assert.equal(path, require.resolve('./fixtures/linked/index'));
         done();

